tests/integration/Product/ProductQueryRequestTest.php

```diff
@@ -304,4 +308,132 @@ function (Product $product) use ($client) {
             }
         );
     }
+
+    public function testLocaleProjectionGetById()
+    {
+        $client = $this->getApiClient();
+
+        ProductFixture::withProduct(
+            $client,
+            function (Product $product) use ($client) {
+                $locale = key($product->getMasterData()->getCurrent()->getName()->toArray());
+
+                $request = RequestBuilder::of()->productProjections()
+                    ->getById($product->getId())
+                    ->localeProjection($locale)
+                    ->staged(true);
+
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
+
+                $localeResult = key($result->getName()->toArray());
+
+                $this->assertInstanceOf(ProductProjection::class, $result);
+                $this->assertSame($locale, $localeResult);
+                $this->assertSame($product->getId(), $result->getId());
+            }
+        );
+    }
+
+    public function testFallbackLocaleProjectionGetById()
+    {
+        $client = $this->getApiClient();
+
+        ProductFixture::withProduct(
+            $client,
+            function (Product $product) use ($client) {
+                $locale = key($product->getMasterData()->getCurrent()->getName()->toArray());
+
+                $request = RequestBuilder::of()->productProjections()
+                    ->getById($product->getId())
+                    ->localeProjection('de')
+                    ->staged(true);
+
+                $response = $this->execute($client, $request);
+                $result = $request->mapFromResponse($response);
+
+                $localeResult = key($result->getName()->toArray());
```

Comment:
This returns the key of an array/map?

Reply:
this is the same logic has my first comment